lib/Yield/Support/SugiyamaStyleGraphLayout/GraphPreparation.cpp

```diff
@@ -229,8 +229,9 @@ partitionLongEdges(InternalGraph &Graph, NodeClassifier<Strategy> &Classifier) {
   revng_assert(Graph.getEntryNode() == nullptr);
   if (EntryNodes.size() > 1) {
     auto EntryPoint = Graph.addNode(nullptr);
-    for (auto &Node : EntryNodes)
-      EntryPoint->addSuccessor(Node, nullptr);
+    for (auto *Node : Graph.nodes())
+      if (!Node->hasPredecessors() && Node->Index != EntryPoint->Index)
```

Comment:
Can you compare pointers instead of `Index`?

Reply:
OK, I don't have enough context, but I trust you.
Consider maybe if it makes sense to implement `operator==` at this point. If not, feel free to mark this as solved.

Reply:
This is one of the core concepts behind how the layouter works. Adding a comparison operator (or a similar alternative) would cause a bit of the rework for everything that touches "internal" (and especially "fake") nodes - and I feel like I'm already somewhat behind the schedule with this PR.

Reply:
Makes sense.